doc/design/docker-rename.md

```diff
@@ -0,0 +1,39 @@
+This is the design proposal for implementing “docker rename” on vSphere Integrated Containers Engine (VIC Engine).
+
+## Design Proposal
+
+### The problem:
+
+Rename involves the containerVM’s display name and the name of the storage folder (and the files in the folder) of the containerVM on the vsphere UI, the container cache of the VIC Engine and the container network configuration (e.g., network alias). Currently both the containerVM's display name and the folder name are created during VM creation using containerName-containerID, which is done for matching the container information obtained from `docker ps` to the VM displayed on the vSphere UI. Renaming the VM display name on the UI can be easily achieved by using govc, which however does not update the storage folder name. In this case, the vi admin would observe inconsistent VM display name and storage folder name, and it becomes difficult the admin to reference the storage folder based on the new VM display name.
+
+### Proposed solution:
+
+- We still use containerName-containerID to assemble the VM name. However, we only use the containerID to set the name of the VM storage folder; therefore, there is no need to worry about the VM display name and storage folder name being inconsistent. 
```

Comment:
I prefer this - there was a reason that it wasn't being done but I don't recall what.
@dougm I *think* this was related only to some of the out-of-band govc testing - do you recall?

If we are doing this I'd prefer that we take the opportunity to move to using `name-shortid` so that we're no longer truncating the human readable name. Ideally we implement this by using a template string as per the constants defined in `lib/config/virtual_container_host.go:35` (add a shortid token). This provides for future requirements of allowing viadmin to apply a naming convention to containerVMs.